### PR TITLE
Features/caris plugin update for hpd4

### PR DESCRIPTION
**Pull request by SamirHassoun**
## SourceDocumentService
App.config, App.config.token, DSC: Added Cuia DB connectionString
CuiaDatabaseHelper.cs: Retrieves next number from [CarisWreckIdSequence] in Cuia DB
CarisController.cs: Added Api controller for retrieving next number from CarisHelper
Program.cs, SourceDocumentController.cs: Improved logging

### CODE REVIEW

**Review by benbhall — APPROVED** [review]

File: src/SourceDocumentService/Controllers/CarisController.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                result = await _cuiaDatabaseHelper.GetNextWreckIdAsync();
+
+                _logger.LogInformation($"{nameof(GetNewWreckageId)}: successfully retrieved next value from wreck id sequence {result}.");
+
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex,$"{nameof(GetNewWreckageId)}: failed to retrieve next value from wreck id sequence.");
+
+                return StatusCode(500, $"Failed to get next number for wreck id: {ex.Message}");
```
Comment (benbhall) [inline src/SourceDocumentService/Controllers/CarisController.cs:42]: We don't to expose exception messages to client in production.

File: src/SourceDocumentService/Program.cs [hunk 2]
```diff
@@ -14,9 +14,15 @@ public static void Main(string[] args)
                 .MinimumLevel.Override("Microsoft", LogEventLevel.Information)
                 .MinimumLevel.Override("Microsoft.AspNetCore", LogEventLevel.Warning)
                 .Enrich.FromLogContext()
-                .WriteTo.File("log.txt", rollingInterval: RollingInterval.Day)
+                .WriteTo.File(
```
Comment (benbhall) [inline src/SourceDocumentService/Program.cs:17]: Guessing this is going to play up in Azure pre and prd for whatever the reason Event service does too?

**Review by benbhall — CHANGES_REQUESTED** [review]

File: src/SourceDocumentService/Program.cs (hunk 2 above)
Comment (SamirHassoun) [reply src/SourceDocumentService/Program.cs:17]: thanks @benbhall,  My part is just to improve the text file logging, fixing it is outside the scope of the story

File: src/SourceDocumentService/Controllers/CarisController.cs (hunk 1 above)
Comment (SamirHassoun) [reply src/SourceDocumentService/Controllers/CarisController.cs:42]: thanks @benbhall, CarisPlugin does not have a proper log file, it uses EventViewer on end-user PC which is not easy to get. So I had no choice put to display the error message

**Review by UKHOGreg — APPROVED** [review]
Looks good!
The only thing I would questions is the lack of unit tests, but you usually add them in a separate PR?

**Review by benbhall — APPROVED** [review]

**Pull request merged**